src/pybind/mgr/dashboard/controllers/oauth2.py

```diff
@@ -0,0 +1,29 @@
+import cherrypy
+
+from dashboard.controllers._auth import ControllerAuthMixin
+from dashboard.exceptions import DashboardException
+from dashboard.services.auth.oauth2 import OAuth2
+
+from .. import mgr
+from . import BaseController, Endpoint, Router
+
+
+@Router('/auth/oauth2', secure=False)
+class Oauth2(BaseController, ControllerAuthMixin):
```

Comment:
I'm curious, why just `BaseController`? Does it generate the correct OpenAPI doc?
```suggestion
class Oauth2(RESTController, ControllerAuthMixin):
```

Reply:
No particular reason other than following the Saml2 implementation and not noticing this.. changed it